Rocket.Chat/Views/Chat/RCTextView.swift

```diff
@@ -72,28 +72,34 @@ class HighlightLayoutManager: NSLayoutManager {
     func updateCustomEmojiViews() {
         customEmojiViews.forEach { $0.removeFromSuperview() }
         customEmojiViews.removeAll()
+        addCustomEmojiIfNeeded()
+    }
 
+    func addCustomEmojiIfNeeded() {
         message?.enumerateAttributes(in: NSRange(location: 0, length: message.length), options: [], using: { attributes, crange, _ in
             if let attachment = attributes[NSAttributedStringKey.attachment] as? NSTextAttachment {
-
-                guard let position1 = textView.position(from: textView.beginningOfDocument, offset: crange.location) else { return }
-                guard let position2 = textView.position(from: position1, offset: crange.length) else { return }
-                guard let range = textView.textRange(from: position1, to: position2) else { return }
-
-                let rect = textView.firstRect(for: range)
-
-                let emojiView = EmojiView(frame: rect)
-                emojiView.backgroundColor = .white
-                emojiView.isUserInteractionEnabled = false
-
-                if let imageUrlData = attachment.contents,
-                    let imageUrlString = String(data: imageUrlData, encoding: .utf8),
-                    let imageUrl = URL(string: imageUrlString) {
-                    emojiView.emojiImageView.sd_setImage(with: imageUrl, completed: nil)
-                    customEmojiViews.append(emojiView)
+                DispatchQueue.main.async {
```

Comment:
By calling it one time per cell reuse, but doing the positioning/adding operations of `EmojiView` on the main thread has the same effect than calling it a lot of times on `layoutSubviews`.